dashboard/app/models/script.rb

```diff
@@ -188,6 +199,22 @@ def self.cache_find_script_level(script_level_id)
     script_level
   end
 
+  # Find the level with the given id from the cache, unless the level build mode
+  # is enabled in which case it is always fetched from the database. If we need to fetch
+  # the level and we're not in level mode (for example because the level was created after
+  # the cache), then an entry for the level is added to the cache.
+  def self.cache_find_level(level_id)
+    level = level_cache[level_id] if self.should_cache?
+
+    # If the cache missed or we're in levelbuilder mode, fetch the level from the db.
```

Comment:
looks like there are about 100 levels that are not in scripts -- I'm not sure how many of them have LevelSources that anyone will reference:

irb(main):002:0* ScriptLevel.distinct(:level_id).count
=> 3903
irb(main):003:0>  Level.count
=> 4009